luda/editor/client/src/timeline/DefaultTrackBody.ts

```diff
@@ -32,22 +49,134 @@ export const DefaultTrackBody: Render<
       onMouseUp(event) {
         if (event.button === MouseButton.right) {
           const clickMs =
-            state.layout.startMs + event.translated.x * state.layout.msPerPixel;
-          state.contextMenu = {
+            state.timelineState.layout.startMs +
+            event.translated.x * state.timelineState.layout.msPerPixel;
+          state.timelineState.contextMenu = {
             type: "trackBody",
             clickMs,
             x: event.x,
             y: event.y,
-            trackId: props.track.id,
+            trackId: state.track.id,
           };
         }
       },
     }),
+    Rect({
+      x: 0,
+      y: props.height / 3,
+      width: props.width,
+      height: props.height / 3,
+      style: {
+        stroke: {
+          color: ColorUtil.Color01(0.25, 0.25, 0.25),
+          width: 1,
+          borderPosition: BorderPosition.inside,
+        },
+      },
+    }),
+    Rect({
+      x: 0,
+      y: 0,
+      width: props.width,
+      height: props.height,
+      style: {
+        stroke: {
+          color: ColorUtil.Black,
+          width: 1,
+          borderPosition: BorderPosition.inside,
+        },
+      },
+    }),
     clips.map((clip) => {
-      return ClipComponent(
-        { timelineState: state, clip },
-        { height: props.height, maxRight: props.width, sashComponent: Sash },
+      if (
+        state.timelineState.actionState?.type === "dragClip" &&
+        clip.id === state.timelineState.actionState.clipId
+      ) {
+        return;
+      }
+      return renderClip(
+        { timelineState: state.timelineState, clip },
+        {
+          height: props.height,
+          maxRight: props.width,
+        },
       );
     }),
+    draggingFakeClip,
   ];
 };
+
+function constrainDraggingClipPlacement(state: TimelineState, clips: Clip[]) {
```

Comment:
I think this logic is only for subtitle. Make this function without `isSubtitleClip` but with some arguments like the minimum gap, or just put this function in SubtitleTrackBody.